### PR TITLE
Merge changes from checklist-converter

### PR DESCRIPTION
**Pull request by theisuru**
see
- ebi-ait/checklist#50 - doc schema store
- ebi-ait/checklist#56 - merge changes back

- Merging changes in ebi-ait/checklist-converter
- Upgrading to spring-boot 3.x
- This is using basic-auth, moving to webin is neccessary
- Biovalidator version upgrade
- Remove unsupported flapdoodle for mongodb test dependency. Integration tests will fail until testcontainers is integrated. Need to run mongo and validator background for test to work.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/biosamples/jsonschemastore/ena/ChecklistConverterService.java [hunk 1]
```diff
@@ -33,25 +34,77 @@ public ChecklistConverterService(SchemaService schemaService, SchemaObjectPopula
...
+            if (textValue.getSynonyms() != null) {
+                values.addAll(textValue.getSynonyms());
+            }
+        }
+        return values;
+    }
+
+    private static Property mapEnaFieldToProperty(Field field) {
+        return new Property(field.getName(), field.getSynonyms(), field.getDescription(), getTypedTemplate(field),
+            field.getUnits(), getCardinality(field.getMandatory()));
```
Comment (amnonkhen) [inline src/main/java/uk/ac/ebi/biosamples/jsonschemastore/ena/ChecklistConverterService.java:64]: Just a vocabulary comment. Cardinality is more appropriate to the "single/multiple" attribute of a field's inclusion in a checklist, i.e. the number of values an attribute can hold for a single entity. 

For required/optional/recommended, my best suggestion is inclusionRequirement or inclusionConstraint.

File: src/test/resources/application-test.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 spring:
   data:
     mongodb:
-      port: 27017
+      port: ${mongodb.container.port}
```
Comment (amnonkhen) [inline src/test/resources/application-test.yml:5]: I don't see the point of using a property placeholder, when you could use the default placeholder `spring.data.mongodb.port`

**Review by amnonkhen — COMMENTED** [review]
added a few comments

File: docker-compose.yml [hunk 3]
```diff
@@ -5,8 +5,7 @@ services:
     ports:
       - 27017:27017
   validator:
-    image: ebispot/elixir-jsonsschema-validator:1.3.0
-    command: npm start
+    image: quay.io/ebi-ait/biovalidator:2.2.2
```
[the remark this answers is not in the document]
Comment (theisuru) [reply docker-compose.yml:8]: Yes
Hopefully it is public.

File: src/main/java/uk/ac/ebi/biosamples/jsonschemastore/service/SchemaService.java [hunk 4]
```diff
@@ -51,7 +50,7 @@ public Optional<JsonSchema> getSchemaById(@NonNull String id) {
         return optionalSchema.map(modelConverter::mongoJsonSchemaToJsonSchema);
     }
 
-    public Page<SchemaOutline> getAllVersionsByAccession(@NotNull String accession, int page, int size) {
+    public Page<SchemaOutline> getAllVersionsByAccession(@NonNull String accession, int page, int size) {
```
[the remark this answers is not in the document]
Comment (theisuru) [reply src/main/java/uk/ac/ebi/biosamples/jsonschemastore/service/SchemaService.java:53]: Let's add these as improvements. (these are not in merge scope)

File: src/main/java/uk/ac/ebi/biosamples/jsonschemastore/service/UserService.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  public UserService(UserRepository userRepository) {
+    this.userRepository = userRepository;
+    passwordEncoder = new BCryptPasswordEncoder();
+  }
+
+  @Override
+  public UserDetails loadUserByUsername(String username) {
+    return userRepository.findByUsername(username)
+        .orElseThrow(() -> new AuthenticationCredentialsNotFoundException("Invalid username or password"));
```
[the remark this answers is not in the document]
Comment (theisuru) [reply src/main/java/uk/ac/ebi/biosamples/jsonschemastore/service/UserService.java:24]: I don't think it's a good security practice to do that.

File: src/test/resources/application-test.yml (hunk 2 above)
Comment (theisuru) [reply src/test/resources/application-test.yml:5]: This is for `testcontainers`. It uses a random port in tests. We need to capture the random port in Spring. I will see if I can directly set spring variable in testcontainer.

**Pull request merged**